src/core/http1/error.rs

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2024 Fastly, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::http1;
+use std::io;
+
+#[derive(Debug)]
+pub enum Error {
+    Io(io::Error),
+    Http(http1::Error),
+    RequestTooLarge(usize),
+    ResponseTooLarge(usize),
+    ResponseDuringContinue,
+    BufferExceeded,
+    Unusable,
+}
```

Comment:
here is a list of RFC 6455 err codes:
```
Protocol Error (1002
Unsupported Data (1003)
Abnormal Closure (1006)
Invalid Frame Payload Data (1007)
Policy Violation (1008)
Message Too Big (1009)
Mandatory Extension (1010)
Internal Server Error (1011)
TLS Handshake (1015)
```
 we might wanna consider adding these as we see fit